libsolidity/analysis/FunctionCallGraph.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolidity/analysis/FunctionCallGraph.h>
+#include <boost/range/adaptor/reversed.hpp>
+
+using namespace std;
+using namespace solidity::frontend;
+
+FunctionCallGraphBuilder::FunctionCallGraphBuilder(ContractDefinition const& _contract):
+	m_contract(&_contract),
+	m_graph(std::make_unique<ContractCallGraph>(_contract))
+{
+	// Create graph for constructor, state vars, etc
+	m_currentNode = SpecialNode::EntryCreation;
+	m_currentDispatch = SpecialNode::InternalCreationDispatch;
+
+	for (ContractDefinition const* contract: _contract.annotation().linearizedBaseContracts | boost::adaptors::reversed)
+	{
+		for (auto const* stateVar: contract->stateVariables())
+			stateVar->accept(*this);
+
+		for (auto arg: contract->baseContracts())
+			arg->accept(*this);
+
+		if (contract->constructor())
+		{
+			add(*m_currentNode, contract->constructor());
+			contract->constructor()->accept(*this);
+			m_currentNode = contract->constructor();
+		}
+	}
+
+	m_currentNode.reset();
+	m_currentDispatch = SpecialNode::InternalDispatch;
+
+	// Create graph for all publicly reachable functions
+	for (auto& [hash, functionType]: _contract.interfaceFunctionList())
+	{
+		(void)hash;
+		if (auto const* funcDef = dynamic_cast<FunctionDefinition const*>(&functionType->declaration()))
+			if (!m_graph->edges.count(funcDef))
+				visitCallable(funcDef);
+
+		// Add all external functions to the RuntimeDispatch
```

Comment:
We don't call this dispatch anymore.

Reply:
We consider the creation graph and the post-deploy graph independent of each other. Both have an "entry node" (the constructor entry point at construction time and the external dispatch after deployment) and both have an internal dispatch (or just "dispatch") node for the indirect function calls.

There is one speciality, though: A function referenced (but not called) at creation time has to be included in the internal dispatch after deployment, because it could have been stored in storage.

Reply:
I think this answer does not match my question :) I was asking about the name of the dispatch that the contract goes through when an external function gets called.

I actually saw #10934 but surprisingly it does not answer my question. Is it `DeploymentDispatch`? `DeployDispatch`?